### PR TITLE
New command in the Azure Storage Contaitner pack for blocking public access to containers

### PR DESCRIPTION
**Pull request by yash-metron**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: link to the issue

## Description
This PR introduces a new command in the Azure Storage Contaitner pack for blocking public access to Azure containers. Initially, I was using a custom script to implement public access prevention in my playbooks. However, following the suggestion by [@Michal Goldshtein](https://paloaltonetworks.enterprise.slack.com/team/WH8QAFMU0), this functionality has now been incorporated directly into the Azure Storage Contaitner pack. This will enable seamless usage of the public access block in various playbooks, reducing the need for custom scripts and improving efficiency.

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by rshunim** [comment]
Hi @yash-metron, thanks for your contribution.
At first, please upgrade the pack version and add a Release Note file.

**Comment by yash-metron** [comment]
Hi @aaron1535 
We are getting these undefined errors from the previous code and not from the changes we have made.
It would be great if you could help me resolving these errors.
Attaching the screenshot of errors from the pre-commit. 
image

**Comment by yash-metron** [comment]
Hi @aaron1535

File: Packs/AzureStorageContainer/pack_metadata.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
     "name": "Azure Storage Container",
     "description": "Create and Manage Azure Storage Container services.",
     "support": "xsoar",
-    "currentVersion": "1.0.22",
+    "currentVersion": "2.0.0",
```
Comment (aaron1535) [inline Packs/AzureStorageContainer/pack_metadata.json:5]: We should leave it with 1.0.23. Please run the `demisto-sdk update-release-notes` command again with the `-u revision` argument

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer_test.py [hunk 2]
```diff
@@ -283,6 +283,42 @@ def test_azure_storage_get_blob_tags_command(requests_mock):
...
+    client = Client(server_url=BASE_URL, verify=False, proxy=False,
+                    account_sas_token=SAS_TOKEN,
+                    storage_account_name=ACCOUNT_NAME, api_version=API_VERSION)
+    result = block_public_access_command(client, {'request_url': url,
+                                                  'headers': headers})
+
+    expected_response = f"Public access to container '{container_name}' has been successfully blocked"
+    assert result.outputs is None
+    assert result.outputs_prefix is None
+    assert result.readable_output == expected_response
```
Comment (aaron1535) [inline Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer_test.py:319]:
```suggestion
    expected_response = f"Public access to container '{container_name}' has been successfully blocked"
    assert result.readable_output == expected_response
```

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer_description.md [hunk 3]
```diff
@@ -41,4 +41,14 @@ Follow one of these steps for authentication based on Azure Managed Identities:
...
+##### Note: This option is relevant only if you want to block public access of a specific container.
+
+Follow these steps to create and copy shared access keys:
+
+1. Navigate to your storage account in the Azure portal.
+2. Under the **Security + networking** section, select the **Access keys** option.
+3. In the **Accesskeys** window, you will get two keys.
+4. Copy any one of them and use it while configuring the instance.
+
+For more information, see [Shared access keys](https://learn.microsoft.com/en-gb/azure/storage/common/storage-account-keys-manage?tabs=azure-portal#regenerate-access-keys).
```
Comment (aaron1535) [inline Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer_description.md:54]:
```suggestion
- For more information, see [Shared access keys](https://learn.microsoft.com/en-gb/azure/storage/common/storage-account-keys-manage?tabs=azure-portal#regenerate-access-keys).

For more information, see [Managed identities for Azure resources](https://learn.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview).
```

**Review by aaron1535 — CHANGES_REQUESTED** [review]
Hi @yash-metron, 
Thank you again for your contribution.
Nice job!
Please pay attention to the required changes and then we can make a demo for the new command you added

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.yml [hunk 4]
```diff
@@ -32,6 +32,10 @@ configuration:
   type: 0
   required: false
   hidden: true
+- display: Shared Key
+  name: shared_key
+  type: 4
+  required: false
```
Comment (aaron1535) [inline Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.yml:38]: Please add `additionalinfo` and `section` to this parameter.

**Comment by aaron1535** [comment]
Hi @yash-metron,
About the errors that you get from the pre-commit
the first one it's in your code. Just remove the unnecessary 'f' string
All other errors are because you deleted the `from CommonServerPython import *` import
If there is no specific reason why you deleted the import, please return it

File: Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py [hunk 5]
```diff
@@ -982,6 +986,80 @@ def generate_sas_token_command(client: Client, args: dict) -> CommandResults:  #
...
+            }
+            response = client.block_public_access(request_url, headers)
+            if response:
+
+                message = f"Public access to container '{container_name}' has been successfully blocked"
+            else:
+                message = f"Failed to block public access, Status code: {response.status_code}"
+        command_results = CommandResults(
+            readable_output=message,
+        )
```
[the remark this answers is not in the document]
Comment (yash-metron) [reply Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py:1057]: No it doesn't exist. It only returns response status and headers.

[the remark this answers is not in the document]
Comment (aaron1535) [reply Packs/AzureStorageContainer/Integrations/AzureStorageContainer/AzureStorageContainer.py:1057]: Okay, in that case you don't need to return the response.
I am trying to understand something in your condition. What is meant 
```
if response:
    message = f"Public access to container '{container_name}' has been successfully blocked"
else:
    message = f"Failed to block public access, Status code: {response.status_code}"
```
in the else scope you get the `response.status_code` **but there is no response**

**Comment by yash-metron** [comment]
Hi @aaron1535 ,
Thanks for the review. I have made changes as per your suggestions.

**Comment by aaron1535** [comment]
Hi @yash-metron,
I see you are still having problems with the `pre-commit`
You can write me on Slack if you want

**Comment by yash-metron** [comment]
> Hi @yash-metron, I see you are still having problems with the `pre-commit` You can write me on Slack if you want

Hi @aaron1535 I couldn't find you on Slack. Could you please share your Slack username so that we can connect there?

**Comment by yash-metron** [comment]
> Hi @yash-metron, I see you are still having problems with the `pre-commit` You can write me on Slack if you want

**Review by aaron1535 — APPROVED** [review]

**Pull request merged**